src/librbd/api/Mirror.cc

```diff
@@ -3032,10 +3099,11 @@ int Mirror<I>::group_promote(IoCtx& group_ioctx, const char *group_name,
   }
 
   cls::rbd::MirrorSnapshotState state;
-  cls::rbd::GroupSnapshotState sync;
-  r = get_last_mirror_snapshot_state(group_ioctx, group_id, &state, &sync);
+  cls::rbd::MirrorGroupSnapshotCompleteState sync;
+  cls::rbd::GroupSnapshotState group_snap_state;
+  r = get_last_mirror_snapshot_state(group_ioctx, group_id, group_snap_state, &state, &sync);
   if (r == -ENOENT) {
-    sync = cls::rbd::GROUP_SNAPSHOT_STATE_INCOMPLETE;
+    sync = cls::rbd::MIRROR_GROUP_SNAPSHOT_INCOMPLETE;
     state = cls::rbd::MIRROR_SNAPSHOT_STATE_PRIMARY_DEMOTED; // XXXMG?
```

Comment:
Since `get_last_mirror_snapshot_state()` now has three output parameters, it would be nice to initialize all three instead of just two of them here.

(I'm not sure what Mykola had in mind though.  If the group doesn't have any mirror snapshots, I don't see how promote can be allowed to do anything -- but that is for a separate discussion.)